### PR TITLE
Prepare RNNT to switch to Numba loss for compatibility

### PR DESCRIPTION
**Pull request by titu1994**
# Changelog
- Add rnnt loss switching capability.
- Enable passing arbitrary kwargs to the inner loss function.
- Add jenkins tests (commented out until 21.04 container is released which supports minimum numba version 0.53 for CUDA 11.2)
- Enable all RNNT model tests using the numba default loss

Signed-off-by: smajumdar <titu1994@gmail.com>

### CODE REVIEW

File: nemo/collections/asr/models/rnnt_models.py [hunk 1]
```diff
@@ -130,6 +136,44 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
             self._optim_variational_noise_std = 0
             self._optim_variational_noise_start = 0
 
+    def _setup_loss(self, cfg: Optional[DictConfig]):
```
Comment (VahidooX) [inline nemo/collections/asr/models/rnnt_models.py:139]: name is a little misleading. How about extract_rnnt_cfg?

File: nemo/collections/asr/models/rnnt_models.py [hunk 2]
```diff
@@ -130,6 +136,44 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
...
+
+                Note that whichever loss_name is selected, that corresponding kwargs will be
+                selected. For the "default" case, the "default_kwargs" will be used, not the
+                "{resolved_default}_kwargs".
+
+        Examples:
+            .. code-block:: yaml
+                loss_name: "default"
+
+                default_kwargs:
```
Comment (VahidooX) [inline nemo/collections/asr/models/rnnt_models.py:158]: What happens if later we change the default rnnt loss later?
is it necessary to have this default?

File: nemo/collections/asr/models/rnnt_models.py [hunk 3]
```diff
@@ -130,6 +136,44 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
...
+                    kwargs2: some_other_val
+
+        Returns:
+            A tuple, the resolved loss name as well as its kwargs (if found).
+        """
+        if cfg is None:
+            cfg = DictConfig({})
+
+        loss_name = cfg.get("loss_name", "default")
+        loss_kwargs = cfg.get(f"{loss_name}_kwargs", None)
```
Comment (VahidooX) [inline nemo/collections/asr/models/rnnt_models.py:171]: Do we need to have the name of the loss in the param?
Is there a way we use target as the class name of the loss and let hydra create the appropriate loss?

File: Jenkinsfile [hunk 4]
```diff
@@ -286,6 +286,44 @@ pipeline {
       }
     }
 
+//     stage('L2: ASR RNNT dev run') {
```
Comment (VahidooX) [inline Jenkinsfile:289]: How about adding a comment why this section is commented out and when it comes back?

**Review by VahidooX — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 4 above)
Comment (titu1994) [reply Jenkinsfile:289]: Its commented out until 21.04 is released, which has Numba 0.53 installed in it. I think well just send a PR next month once we get that container.

File: nemo/collections/asr/models/rnnt_models.py (hunk 1 above)
Comment (titu1994) [reply nemo/collections/asr/models/rnnt_models.py]: Done

File: Jenkinsfile (hunk 4 above)
Comment (titu1994) [reply Jenkinsfile:295]: Done

**Review by VahidooX — APPROVED** [review]

**Pull request merged**